zerver/openapi/zulip.yaml

```diff
@@ -4425,6 +4425,104 @@ paths:
+                                  type: array
+                                  description: |
+                                    An array containing objects for the newly created scheduled messages.
+                                  items:
+                                    $ref: "#/components/schemas/ScheduledMessage"
+                              example:
+                                {
+                                  "type": "scheduled_messages",
+                                  "op": "add",
+                                  "scheduled_messages":
+                                    [
+                                      {
+                                        "scheduled_message_id": 17,
+                                        "type": "private",
+                                        "to": [6],
+                                        "topic": "",
+                                        "content": "Hello there!",
+                                        "rendered_content": "<p>Hello there!</p>",
+                                        "deliver_at": "2023-04-19 18:24:56",
+                                      },
+                                    ],
+                                }
+                            - type: object
+                              additionalProperties: false
+                              description: |
+                                Event containing details for an edited scheduled message.
+                              properties:
+                                id:
+                                  $ref: "#/components/schemas/EventIdSchema"
+                                type:
+                                  allOf:
+                                    - $ref: "#/components/schemas/EventTypeSchema"
+                                    - enum:
+                                        - scheduled_messages
+                                op:
+                                  type: string
+                                  enum:
+                                    - update
+                                scheduled_message:
+                                  $ref: "#/components/schemas/ScheduledMessage"
+                              example:
+                                {
+                                  "type": "scheduled_messages",
+                                  "op": "update",
+                                  "scheduled_message":
+                                    {
+                                      "scheduled_message_id": 17,
+                                      "type": "private",
+                                      "to": [6],
+                                      "topic": "",
+                                      "content": "Hello there!",
+                                      "rendered_content": "<p>Hello there!</p>",
+                                      "deliver_at": "2023-04-19 18:24:56",
+                                    },
```

Comment:
Please post the event format in "#api design" for discussion, maybe in its own topic.

Reply:
Started an api design conversation about the event formats: https://chat.zulip.org/#narrow/stream/378-api-design/topic/schedule.20messages.3A.20event.20formats/near/1556133